### PR TITLE
Remove __dict__ override and refactor code

### PR DESCRIPTION
**Pull request by KomalBhadani27**
- Remove __dict__ override
- Add convert_to_dict methods for Website and Contents for logging purpose
- Refactor code in `stamp_generation/` to use class objects instead of dictionary

### CODE REVIEW

File: stamp_generation/stamp_generator.py [hunk 1]
```diff
@@ -19,8 +19,8 @@ def generate_stamp(self):
...
-            publisher_domain=self._website['domain'],
-            logo_url=self._website['logo_url'],
-            canonical_url=self._website['url'],
-            contents=self._website['contents']['content_list'],
-            pages=self.stamp_pages)
+            publisher_domain=self._website.domain,
+            logo_url=self._website.logo_url,
+            canonical_url=self._website.url,
+            contents=self._website.contents.content_list,
+            pages=self.stamp_pages.stamp_pages)
```
[the remark this answers is not in the document]
Comment (KomalBhadani27) [reply stamp_generation/stamp_generator.py:26]: These are the variable names that will be used in Jinja Template. 
https://en.wikipedia.org/wiki/Jinja_(template_engine)

File: data_models/contents.py [hunk 2]
```diff
@@ -14,8 +14,21 @@ def add_content(self, _content):
         """This method appends extracted content to content_list"""
 
         self.content_list.append(_content)
-        self.__dict__['content_list'].remove(_content)
-        self.__dict__['content_list'].append(_content.__dict__)
+
+    def convert_to_dict(self):
+        formatted_contents_list = list()
+        formatted_contents = dict()
```
[the remark this answers is not in the document]
Comment (KomalBhadani27) [reply data_models/contents.py]: Updated accordingly!

File: data_models/contents.py [hunk 3]
```diff
@@ -14,8 +14,21 @@ def add_content(self, _content):
...
+
+        for _content in self.content_list:
+            formatted_contents_list.append(_content.__dict__)
+
+        for key, value in self.__dict__.items():
+            if key == 'content_list':
+                formatted_contents[key] = formatted_contents_list
```
[the remark this answers is not in the document]
Comment (KomalBhadani27) [reply data_models/contents.py]: Thanks for suggestion! Updated.

File: stamp_generation/stamp_generator.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (detel) [reply stamp_generation/stamp_generator.py:26]:
```suggestion
            self._website.domain,
            self._website.logo_url,
            self._website.url,
            self._website.contents.content_list,
            self.stamp_pages.stamp_pages)
```

[the remark this answers is not in the document]
Comment (KomalBhadani27) [reply stamp_generation/stamp_generator.py:26]: Will be considered in later iterations.

**Pull request merged**